feat(go): add metrics docs (#16073)
getsentry/sentry-docs: platform-includes/metrics/usage/go.mdx

> Once you initialize the SDK, you can create a meter and send metrics using its methods.

Reviewer: ```suggestion
Once the SDK is initialized, you can record metrics using a meter.
```

> The meter exposes three methods that you can use to capture different types of metric information: `Count`, `Gauge`, and `Distribution`.

Reviewer: ```suggestion
A meter supports three types of measurements: `Count`, `Gauge`, and `Distribution`. Each is suited to a different kind of data.
```

This and the suggestion above are opinionated, but I think it helps make this feel a little less mechanical.
